scipy/stats/_continuous_distns.py

```diff
@@ -604,6 +605,18 @@ def _cdf(self, x, a, b):
     def _ppf(self, q, a, b):
         return sc.btdtri(a, b, q)
 
+    def _sf(self, x, a, b):
+        return _lazywhere(x < 0.5,
```

Comment:
`_lazywhere` is moderately slow so I try to avoid it. you're inverting the order to increase precision correct? Can this optimization be pushed down into btdtr?

Reply:
> Are you inverting the order of a,b to increase precision?

Yes.

> Can this optimization be pushed down into btdtr? 

@WarrenWeckesser what do you think? 

> Is this optimization needed?

It doesn't really fix any bugs (e.g. gh-12635, gh-12794), so I could go either way.

Reply:
@sethtroisi, I agree `_lazywhere` is slow, but the first goal here is improved accuracy.  Eventually we can convert many of the pdf/cdf/etc calculations to compiled code (see, for example, https://github.com/scipy/scipy/pull/13328), but it is OK to make improvements in stages.

Here's another example where this change improves the accuracy of the result.  Using mpsci (based on mpmath), this code

    import mpmath
    mpmath.mp.dps = 80

    from mpsci.distributions import beta
    print(float(beta.sf(0.99999, 2, 3))

prints

    3.999969999945388e-15

So in SciPy, beta.isf(3.999969999945388e-15, 2, 3) should
be close to 0.99999.

With SciPy 1.5.2:

    In [5]: beta.isf(3.999969999945388e-15, 2, 3)
    Out[5]: 0.9999900026399827

With this PR:

    In [10]: beta.isf(3.999969999945388e-15, 2, 3)
    Out[10]: 0.99999